File: app/src/main/java/com/ae/apps/pnrstatus/exceptions/StatusException.java
/*
 * Copyright 2013 Midhun Harikumar
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.ae.apps.pnrstatus.exceptions;

/**
 * Denotes an exception that occured while checking for status
 * 
 * @author midhun_harikumar
 * 
 */
public class StatusException extends Exception {

	private static final long	serialVersionUID	= 4678860372933762653L;

	public enum ErrorCodes {
		PARSE_ERROR, NETWORK_ERROR, EMPTY_RESPONSE, URL_ERROR
	}

	private ErrorCodes	errorCode;

	public StatusException() {
		super();
	}

	public StatusException(String detailMessage, Throwable throwable) {
		super(detailMessage, throwable);
	}

	public StatusException(String detailMessage) {
		super(detailMessage);
	}

	public StatusException(String message, ErrorCodes code) {
		super(message);
		errorCode = code;
	}

	public StatusException(Throwable throwable, ErrorCodes code) {
		super(throwable);
		errorCode = code;
	}

	public StatusException(String message, Throwable throwable, ErrorCodes code) {
		super(message, throwable);
		errorCode = code;
	}

	public ErrorCodes getErrorCode() {
		return errorCode;
	}

}

File: app/src/main/java/com/ae/apps/pnrstatus/exceptions/StatusException.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Midhun Harikumar
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.ae.apps.pnrstatus.exceptions

/**
 * Denotes an exception that occured while checking for status
 *
 * @author midhun_harikumar
 */
class StatusException : Exception {
    enum class ErrorCodes {
        PARSE_ERROR, NETWORK_ERROR, EMPTY_RESPONSE, URL_ERROR
    }

    var errorCode: ErrorCodes? = null
        private set

    constructor() : super()

    constructor(detailMessage: String?, throwable: Throwable?) : super(detailMessage, throwable)

    constructor(detailMessage: String?) : super(detailMessage)

    constructor(message: String?, code: ErrorCodes?) : super(message) {
        errorCode = code
    }

    constructor(throwable: Throwable?, code: ErrorCodes?) : super(throwable) {
        errorCode = code
    }

    constructor(message: String?, throwable: Throwable?, code: ErrorCodes?) : super(
        message,
        throwable
    ) {
        errorCode = code
    }

    companion object {
        private const val serialVersionUID = 4678860372933762653L
    }
}
